### PR TITLE
Ec dh general definitions doc

### PR DESCRIPTION
**Pull request by ilagordhan**
New Documents for Orientation and Implementation

### CODE REVIEW

File: content/data-hub/general/data-hub-concepts.md [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+
+Describes concepts behind using the Mendix Data Hub and sharing datasets.
+
+Data Hub makes it simpler to use data between Apps. Catalog allows governance of which data is used by whom, and life cycle management informing people of changes, and with good versioning of OData contracts.
+
+##  1.1 The Role of a Data Hub in an Organization
+
+The Catalog is the hub in the collaboration, discovery and connection of an organizations data source.   It catalogs microservices from the organization's business applications systems using OData contracts. The Data Hub can already import OData contracts from 3rd party systems such as Teamcenter, SAP & Microsoft and OData v4services from other business applications.
+
+The Data Hub facilitates an integration mechanism, where architects and lead developers can define data that is available and what the data means. Citizen Developers?? can use the data easily, ??and where impact analysis for who uses which data is out of the box??.
```
Comment (juuulka) [inline content/data-hub/general/data-hub-concepts.md:22]:
```suggestion
The Data Hub facilitates an integration mechanism, where architects and lead developers can define data that is available and what the data means. Citizen Developers?? can use the data easily, ??and where impact analysis for who uses which data is out of the box?.
```

File: content/data-hub/general/data-hub-concepts.md [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+
+# 1 Introduction
+
+This document describes the concepts behind using the Mendix Data Hub and sharing datasets in your app development.
+
+Catalog allows governance of which data is used by whom, and life cycle management informing people of changes, and with good versioning of OData contracts.
+
+##  1.1 The Data Hub Components
+
+Data Hub is made up of the following components:
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:24]: These are parts of the Catalog UI. The Data Hub includes the APIs that connect to Studio and Studio Pro.

File: content/data-hub/general/data-hub-concepts.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+
+# Supported OData Protocols
+
+OData v3 (Mendix only)
+
+OData v4 (Other Business applications)
+
+## Service Versions
+
+Every published OData service  has a version number, and apps that consume a datasource will consume from a specific version. Updates and changes to a service will be indicated by a change in the version number if good practice is followed by the data source originators. This means that there may be several versions of a registered data source available in the catalog and they will all be listed as separate items in the search results for the same-named data source.
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:325]:
```suggestion
Every published OData service  has a version number, and apps that consume a data source will consume from a specific version. Updates and changes to a service will be indicated by a change in the version number if good practice is followed by the data source originators. This means that there may be several versions of a registered data source available in the catalog and they will all be listed as separate items in the search results for the same-named data source.
```

File: content/data-hub/general/data-hub-concepts.md [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+
+ Consume is of a specific dataset (entityset), of an endpoint, in a particular environment.
+
+ 
+
+In Studio Pro, external data sources registered in the Data Hub Catalog and can be discovered through the **Data Hub Pane**  and introduced in the domain model as *external entities* which can be used with local entities with the difference that external entities **connect** to data that is maintained in the source application and therefore is **read or consumed**. Changes to this data is done in the originating apps.
+
+
+
+## Entitysets
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:265]:
```suggestion
## Entity sets
```

File: content/data-hub/general/data-hub-concepts.md [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+
+To make data from your apps available to others, you can publish the datasets in an **OData service** and register it in the Data Hub. In a Mendix app, the datasets are the **Entity sets** for a defined **Entity**. Further details such as the location of the registered dataset are also captured: for Mendix app deploying to the Mendix cloud this information is taken during the deployment process.
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:267]:
```suggestion
To make data from your apps available to others, you can publish the datasets in an **OData service** and register it in the Data Hub. In a Mendix app, the datasets are the **Entity sets** for a defined **Entity**. Further details such as the location of the registered dataset are also captured: for a Mendix app deploying to the Mendix Cloud this information is captured during the deployment process. 
```

File: content/data-hub/general/data-hub-concepts.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+
+{{% alert type="info" %}}
+This is not compulsory, and users can continue to use an older version of a service unless the new version was deployed to the same service endpoint as the previous version. In Studio Pro, new versions of a service are indicated and users can choose to **Update** the service, or **Switch** to another version of the service deployed to another endpoint.
+{{% /alert %}}
+
+It is good practice that publishers of a service serve a notice of deprecation on a service version that will be replaced with a new service that may contain breaking changes which would cause the consuming app to fail. In this case the updated service should be deployed to a new service endpoint. In this case, in Studio Pro, users will get the option to **Switch** to the new version. 
+
+# **Registration**
+
+This is the process of “adding” a shareable asset – data source (service) or dataset – to Data Hub. This is means that the metadata is available and can be searched/curated etc.
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:247]:
```suggestion
This is the process of “adding” a shareable asset – data source (service) or dataset (entity) – to Data Hub. This is means that the metadata is available and can be searched/curated etc.
```

File: content/data-hub/general/data-hub-concepts.md [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+
+It is the endpoint.
+
+# Datasets
+
+To users, the **dataset**  data, regardless of the system “owning” the data.  A **dataset** is the data similar to a table or excel sheet, with the actual set of data that the user will connect to.
+
+An **entity** is the modelling construct for a dataset - or more specifically in Mendix, the **entityset** - which is the connection to the available data.
+
+The **external entity** in the consuming app connects to the dataset that is maintained in the originating app via the published service (the API to the originating app) registered in the DHC. Currently datasets for external entities are updated in the source application of the data source. Data is retrieved on the initiative by the consuming app (pull-based), so Data Hub does not play a role in updating the consuming apps: the consuming apps receive the latest data when they “request it” through the API.
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:231]:
```suggestion
The **external entity** in the consuming app connects to the dataset that is maintained in the originating app via the published service (the API to the originating app) registered in the Data Hub Catalog. Datasets for external entities are updated in the source application of the data source. Data is retrieved on the initiative by the consuming app (pull-based), so Data Hub does not play a role in updating the consuming apps: the consuming apps receive the latest data when they request it through the API.
```

File: content/data-hub/general/data-hub-concepts.md [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+
+External entities are read-only, so it is not possible to change te structural values of attributes or associations between two external entities.
+
+For full details see [external entities](ref??)
+
+
+
+# Data Sources 
+
+Data sources are the apps/services which define the data that is being made available through it. The generic term used for the app, database or service which owns and provides access to a dataset that can be consumed in Studio/Studio Pro through external entities. From the point of view of the Studio/Studio Pro user, they are connecting through a data source when they use the external entity of this source and using the dataset of the selected entity.
```
Comment (ezagroba) [inline content/data-hub/general/data-hub-concepts.md:219]: ```Data sources are the apps/services``` can we say they're the services instead?

Comment (ilagordhan) [reply content/data-hub/general/data-hub-concepts.md:219]: Thanks for the feedback. The doc is very much is rough draft stage and not ready for review.

**Pull request closed without merging**